### PR TITLE
py-sphinx-rtd-theme: add 1.2.2 and py-sphinxcontrib-jquery: add 4.1

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/readthedocs/sphinx_rtd_theme/tree/1.2.2
https://github.com/sphinx-contrib/jquery/tree/v4.1

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py [hunk 1]
```diff
@@ -12,8 +12,11 @@ class PySphinxcontribJquery(PythonPackage):
     homepage = "https://github.com/sphinx-contrib/jquery"
     pypi = "sphinxcontrib-jquery/sphinxcontrib-jquery-2.0.0.tar.gz"
 
+    version("4.1", sha256="1620739f04e36a2c779f1a131a2dfd49b2fd07351bf1968ced074365933abc7a")
     version("2.0.0", sha256="8fb65f6dba84bf7bcd1aea1f02ab3955ac34611d838bcc95d4983b805b234daa")
 
-    depends_on("python@3.5:", when="@2:", type=("build", "run"))
-    depends_on("py-setuptools@65:", type=("build", "run"))
     depends_on("py-flit-core@3.7:", when="@3:", type="build")
+    depends_on("py-sphinx@1.8:", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py:19]: Although the `py-sphinx` dependency was only added for versions `@4:` it is clearly a dependency for all versions.

File: var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py [hunk 2]
```diff
@@ -12,8 +12,11 @@ class PySphinxcontribJquery(PythonPackage):
...
+
+    # Historical dependencies
+    depends_on("py-setuptools@65:", when="@:3", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py:22]: Any idea where 65+ comes from?

Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py:22]: Not really, but I suspect that the one who added it wanted to add some restriction to make sure that `setuptools` can handle installation via `pyproject.toml`

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py:22]: Oh, it's right there in the `pyproject.toml`, not sure how I missed that. Thanks!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**